File: src/app/models/AlertRule.js


// Define collection and schema for AlertRule
export interface AlertRule {
    name:
	type : string
    expression:
	type : string
    Tenant:
	type : Schema.Types.ObjectId
    Streams:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TelemetryStream' }]
    Alerts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Alert' }]
    Severity:
 	type : String
#
    collection: 'alertRules'
}

File: iotonaspdotnet/Domain/AlertRule.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class AlertRule
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long AlertruleId { get; set; }
 public virtual string Name { get; set; }
 public virtual string Expression { get; set; }
public virtual Tenant Tenant { get; set; }
public virtual TelemetryStream Streams { get; set; }
public virtual Alert Alerts { get; set; }
 public virtual AlertSeverity Severity { get; set; }

    public static AlertRule FromRequest(AlertRuleRequest request) {
        return new AlertRule {
            Id = request.Id,
            Name = request.Name,
            Expression = request.Expression,
            Severity = request.Severity,
        };
    }
}
